Sources/Build/BuildPlan.swift

```diff
@@ -1251,6 +1320,200 @@ public final class SwiftTargetBuildDescription {
+                VFSOverlay.Directory(
+                    name: buildArtifactDirectory.pathString,
+                    contents: [
+                        VFSOverlay.File(
+                            name: moduleMapFilename,
+                            externalContents:
+                                buildArtifactDirectory.appending(component: unextendedModuleMapFilename).pathString
+                        )
+                    ]
+                )
+            ]).write(to: unextendedModuleMapOverlayPath, fileSystem: fileSystem)
+
+            try VFSOverlay(roots: [
+                VFSOverlay.Directory(
+                    name: publicHeadersPath.pathString,
+                    contents:
+                        // Public headers
+                        try Set(fileSystem.getDirectoryContents(publicHeadersPath)
+                                .map(publicHeadersPath.appending(component:)))
+                                .filter(publicHeadersPath.isAncestor(of:))
+                                .map { headerPath in
+                                    VFSOverlay.File(
+                                        name: headerPath.basename,
+                                        externalContents: headerPath.pathString
+                                    )
+                                }
+                ),
+                VFSOverlay.Directory(
+                    name: buildArtifactDirectory.pathString,
+                    contents: [
+                        // Module map
+                        VFSOverlay.File(
+                            name: moduleMapFilename,
+                            externalContents:
+                                buildArtifactDirectory.appending(component: moduleMapFilename).pathString
+                        )
+                    ]
+                ),
+                VFSOverlay.Directory(
+                    name: buildArtifactDirectory.pathString,
+                    contents: [
+                        // Generated $(ModuleName)-Swift.h header
+                        VFSOverlay.File(
+                            name: generatedInteropHeaderPath.basename,
+                            externalContents: generatedInteropHeaderPath.pathString
+                        )
+                    ]
+                ),
+            ]).write(to: allProductHeadersPath, fileSystem: fileSystem)
+
+            swiftTargetBuildDescription.additionalFlags += [
+                // Builds Objective-C portion of module.
+                "-import-underlying-module",
+                // Pass VFS overlay to the underlying Clang compiler.
+                "-Xcc",
+                "-ivfsoverlay",
+                "-Xcc",
+                allProductHeadersPath.pathString,
+                // Pass VFS overlay to the underlying Clang compiler.
+                "-Xcc",
+                "-ivfsoverlay",
+                "-Xcc",
+                unextendedModuleMapOverlayPath.pathString
```

Comment:
nit/optional: Something you could potentially consider is a helper method on `SwiftTargetBuildDescription` for appending Clang flags. It could add `"-Xcc"` before appending each value to `additionalFlags`.

Reply:
Done. Good call out!